js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/AssetChecksStatusSummary.tsx

```diff
@@ -51,26 +63,70 @@ const AssetCheckIconsOrdered: {
   },
 ];
 
+const titlePerCheckType = (type: AssetCheckIconType) => {
+  switch (type) {
+    case AssetCheckExecutionResolvedStatus.IN_PROGRESS:
+      return 'In progress';
+    case AssetCheckExecutionResolvedStatus.SKIPPED:
+      return 'Skipped';
+    case AssetCheckExecutionResolvedStatus.SUCCEEDED:
+      return 'Succeeded';
+    case 'NOT_EVALUATED':
+      return 'Not evaluated';
+    case 'WARN':
+      return 'Failed';
+    case 'ERROR':
+      return 'Failed';
+    default:
+      assertUnreachable(type);
+  }
+};
+
+const ChecksSummaryPopover = ({
+  type,
+  assetKey,
+  assetChecks,
+}: {
+  type: AssetCheckIconType;
+  assetKey: AssetKey;
+  assetChecks: AssetCheckLiveFragment[];
+}) => {
+  return (
+    <Box flex={{direction: 'column'}} style={{maxHeight: 300, overflowY: 'auto'}}>
+      <Box padding={{horizontal: 12, vertical: 8}} border="bottom">
+        <Subtitle2>{`${titlePerCheckType(type)} checks`}</Subtitle2>
+      </Box>
+      {assetChecks.map((check) => (
+        <CheckStatusRow key={check.name} assetCheck={check} assetKey={assetKey} />
+      ))}
+    </Box>
+  );
+};
+
+function statusFromCheck(check: AssetCheckLiveFragment): AssetCheckIconType {
```

Comment:
It might be nice to call this `iconTypeFromCheck` since the checks have a `status` already.

Sorry this icon type thing is kinda layering additional logic in here, I think this was my doing...